### PR TITLE
[JCLOUDS-841] AffinityGroups operations implemented

### PR DESCRIPTION
**Pull request by ilgrosso**

### CODE REVIEW

File: azurecompute/src/test/java/org/jclouds/azurecompute/features/AffinityGroupApiLiveTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      final UpdateAffinityGroupParams params = UpdateAffinityGroupParams.builder().
+              description(GROUP_NAME + " description").
+              build();
+
+      final String requestId = api().update(GROUP_NAME, params);
+      assertTrue(operationSucceeded.apply(requestId), requestId);
+   }
+
+   @Test(dependsOnMethods = "testUpdate")
+   public void testDelete() throws Exception {
```
Comment (nacx) [inline azurecompute/src/test/java/org/jclouds/azurecompute/features/AffinityGroupApiLiveTest.java]: Should this better be an `@AfterClass` so we also delete the group if the update test fails?

File: azurecompute/src/test/java/org/jclouds/azurecompute/features/AffinityGroupApiMockTest.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      try {
+         final AffinityGroupApi api = api(server.getUrl("/")).getAffinityGroupApi();
+
+         assertEquals(api.delete("mygroup"), "request-3");
+
+         assertSent(server, "DELETE", "/affinitygroups/mygroup");
+      } finally {
+         server.shutdown();
+      }
+   }
```
Comment (nacx) [inline azurecompute/src/test/java/org/jclouds/azurecompute/features/AffinityGroupApiMockTest.java]: Add mock tests that exercise the fallbacks you explicitly define in the API class.

**Comment by nacx** [comment]
Just a couple minors. lgtm! Thanks @ilgrosso. Once addressed we'll have to wait until 1.9.0 is released to merge this.

**Comment by ilgrosso** [comment]
Thx for reviewing @nacx - the additional commit should address your review, please let me know.

**Comment by nacx** [comment]
Thanks @ilgrosso! Can you squash the commits so I can cleanly merge the it?

**Comment by ilgrosso** [comment]
@nacx commits squashed, thanks!

**Comment by nacx** [comment]
Pushed to [master](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/d928dbcc) and [1.9.x](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/15fcd411). Thanks!

**Pull request closed without merging**